app/models/assignment.rb

```diff
@@ -896,6 +900,24 @@ def can_upload_starter_code?
     false
   end
 
+  # Returns true if this is a peer review, meaning it has a parent assignment,
+  # false otherwise.
+  def is_peer_review?
+    not self.parent_assignment_id.nil?
+  end
+
+  # Checks if this assignment has any children which reference this assignment
+  def has_peer_review_assignments?
+    not Assignment.where(:parent_assignment_id => self.parent_assignment_id).empty?
+  end
```

Comment:
That's pretty interesting. I don't think it's redundant, but I don't really like the approach. (Also I think the id should be passed in, not parent_assignment_id)

Why don't we create another association `has_one :peer_review` which is the inverse of `parent_assignment`, and check whether that's nil.